lib/src/state/design.dart

```diff
@@ -1464,7 +1480,7 @@ abstract class Design with UnusedFields implements Built<Design, DesignBuilder>,
     }
   }
 
-  ListBuilder<Mismatch> _find_mismatches_on_substrand(Domain substrand) {
+  ListBuilder<Mismatch> _find_mismatches_on_substrand(Domain substrand, {bool only_insertion_deletion = false}) {
```

Comment:
I think I'm a bit confused what's happening here. The class `Mismatch` already exists, and so does this private method `_find_mismatches_on_substrand`, is that correct? If I recall `Mismatch` specifically refers to a mismatching DNA base pair, e.g., an A next to a G.

A new class should be created that specifically represents an unpaired deletion or insertion. In fact, please rename the actions and field names above to use the phrase "unpaired" instead of "mismatch". The concepts are slightly different so it's good to use a different word. For example, a DNA base on one strand does not mismatch the adjacent strand if the adjacent strand has no DNA sequence assigned, but this would be "unpaired" if we were talking about a deletion.

More importantly, the class `Mismatch` was designed specifically for DNA bases: it has a field `int within_insertion` that refers to the mismatching base being within an insertion in the domain, whereas no unpaired deletion could be "within" an insertion. Since the `Mismatch` class doesn't model the data we want to describe for unpaired deletions/insertions, it should not be used. Let's make a brand new class.

It might be a good idea to rename `Mismatch` to `DNABasePairMismatch` to make it clear that this is the only thing it refers to.

Now that I think about it, we don't even need a new state class to represent unpaired insertions/deletions. The class `Address` already captures exactly what we want. In other words I think the memoized getter above `insertion_deletion_mismatches_map` should, first, be renamed `unpaired_insertion_deletion_map`, then have its type be `BuiltMap<Domain, BuiltList<Address>>`. In other words each `Domain` just maintains a list of `Address`'es of unpaired insertions/deletions. The view code can use this to figure out where to put the red stars.

This might require a bit of refactoring to create a common View component that can be used for all three purposes: DNA base mismatches, Domain name mismatches, and unpaired insertions/deletions.